### PR TITLE
Shl edit samrecord docs

### PR DESCRIPTION
**Pull request by sooheelee**
Clarify SAMRecord documentation so that users are not confused by thinking the opposite of what is meant.

- Three variable descriptions changed: getAlignmentStart, setAlignmentStart and getAlignmentEnd. Before, each read, "position of the clipped sequence," and now each reads, "position of the sequence remaining after clipping," for all three variable descriptions. 
- In addition, took liberty to clarify "or 0 if there is no position" further. Now reads, "If no position, as would occur e.g. for the unmapped mate of a singly mapping pair, returns 0." Talked briefly to LB and agreed _not_ to link to GATK documentation within htsjdk given it is a standalone project.

### CODE REVIEW

**Comment by sooheelee** [comment]
The commit with comments contain the actual documentation updates. The later commit (with no comments, sorry about that) just formats line spacing. Newbie here so please forgive.

**Comment by nh13** [comment]
I am confused by the second line in each (singly mapped pair).  I am not sure about the need for this change, but if you are adamant, then how about: `1-based inclusive leftmost position of the sequence remaining after clipping, or 0 if there is no position (ex. the read is unmapped)`

**Comment by sooheelee** [comment]
@nh13, I believe the top of the SAMRecord doc talks about the types of unmapped reads that the GATK documentation refer to as part of a singly mapping pair [here](https://software.broadinstitute.org/gatk/guide/article?id=6976). I don't think the SAMRecord doc means to imply any unmapped read. My conclusion is that it must refer to those with coordinates within the SAM record but with `*` in the CIGAR field as happens for unmapped reads of singly mapping pairs.

Please correct if I'm mistaken.

If you feel strongly about merging the two sentences, and keeping it simple, then I can do:

> 1-based inclusive leftmost position of the sequence remaining after clipping, or 0 if there is no position, e.g. for an unmapped read in a singly mapping pair.

**Comment by julianhess** [comment]
It will return 0 for a totally unmapped read — it simply regurgitates what's in the BAM.  According to SAM spec, a totally unmapped read will have `RNAME = *` and `POS = 0`.  I agree with @nh13's more general wording.

**Comment by lbergelson** [comment]
@vdauwera The newlines don't make any difference to the doc output.

**Comment by nh13** [comment]
@sooheelee let me know when you make the change and I can look again.

File: src/main/java/htsjdk/samtools/SAMRecord.java [hunk 1]
```diff
@@ -572,14 +572,16 @@ protected static String resolveNameFromIndex(final int referenceIndex, final SAM
     }
 
     /**
-     * @return 1-based inclusive leftmost position of the clipped sequence, or 0 if there is no position.
+     * @return 1-based inclusive leftmost position of the sequence remaining after clipping. 
+     * If no position, as would occur e.g. for the unmapped mate of a singly mapping pair, returns 0.
```
Comment (droazen) [inline src/main/java/htsjdk/samtools/SAMRecord.java]: singly mapping -> singly-mapped (here and below

**Comment by sooheelee** [comment]
@nh13 Now reads "1-based inclusive leftmost position of the sequence remaining after clipping, or 0 if there is no position, e.g. for unmapped read."

**Comment by sooheelee** [comment]
Thanks for the feedback @julianhess @nh13

**Comment by vdauwera** [comment]
Sounds like we have 👍 from everyone then?

**Comment by lbergelson** [comment]
Congrats @sooheelee on your first htsjdk commit!

**Comment by sooheelee** [comment]
Thank you @lbergelson!

**Pull request merged**